### ISSUE TITLE
walrus operator

### ISSUE THREAD

**Issue by Akuli** [Akuli/jou#284]
In new versions of python you can do `while foo := some_function()`. In C, this is very common, e.g. `while ((c = getchar()) != EOF)`.

Jou should also have a `:=` operator. This is even more important for Jou than for Python, because Jou aims to be similar to C.

**Comment by Moosems** [comment]
Yes please! I like how python uses parentheses for it to show it is a complete and separate statement. One thing I'd make possible in Jou is allow it to set attributes (I.E. `while ((self.foo := bar) == bar): ...`).
